### ISSUE TITLE
Add 'show-in-file-manager' to atom-pane

### ISSUE THREAD

**Issue by lengbing** [atom/tree-view#589]
Would you like to add commond 'show-in-file-manager' to atom-pane ? I guess it would be nice .

**Comment by Tyriar** [comment]
+1 Sublime has this and I used it (right click text editor -> Open containing folder).

The alternative is to right click the text editor, select reveal in tree view and then right click the file in the tree view to open it in the file manager.

**Comment by liuderchi** [comment]
+1. Totally agree with @Tyriar. 
Also the *alternative* @Tyriar mentioned above may fail if the text file is not in the tree view

**Comment by nemonein** [comment]
+1

**Comment by tommiller1** [comment]
any plugin that adds this behaviour?

**Comment by liuderchi** [comment]
Temporarily solution :

1. paste the code below into your `~/.atom/init.coffee` file:
(use _Edit (or File) > Init Script_ menu to open `init.coffee`)
2. restart atom editor
3. would generate a menu item _Show In File Manager_ when right-click on editor tab

```coffee
atom.commands.add 'atom-text-editor', 'util:show-file-in-manager', ->
  return unless editor = atom.workspace.getActiveTextEditor()
  path = require 'path'
  {BufferedProcess} = require 'atom'

  # NOTE ref: tree-view.coffee @line:486
  showSelectedEntryInFileManager = (path) ->
    {command, args, label} = fileManagerCommandForPath(path, true)

    handleError = (errorMessage) ->
      atom.notifications.addError "Opening #{if isFile then 'file' else 'folder'} in #{label} failed",
        detail: errorMessage
        dismissable: true

    errorLines = []
    stderr = (lines) -> errorLines.push(lines)
    exit = (code) ->
      failed = code isnt 0
      errorMessage = errorLines.join('\n')

      # Windows 8 seems to return a 1 with no error output even on success
      if process.platform is 'win32' and code is 1 and not errorMessage
        failed = false

      handleError(errorMessage) if failed

    showProcess = new BufferedProcess({command, args, stderr, exit})
    showProcess.onWillThrowError ({error, handle}) ->
      handle()
      handleError(error?.message)

  # NOTE ref: tree-view.coffee @line:459
  fileManagerCommandForPath = (pathToOpen, isFile) ->
    switch process.platform
      when 'darwin'
        command: 'open'
        label: 'Finder'
        args: ['-R', pathToOpen]
      when 'win32'
        args = ["/select,\"#{pathToOpen}\""]

        if process.env.SystemRoot
          command = path.join(process.env.SystemRoot, 'explorer.exe')
        else
          command = 'explorer.exe'

        command: command
        label: 'Explorer'
        args: args
      else
        # Strip the filename from the path to make sure we pass a directory
        # path. If we pass xdg-open a file path, it will open that file in the
        # most suitable application instead, which is not what we want.
        pathToOpen =  path.dirname(pathToOpen) if isFile

        command: 'xdg-open'
        label: 'File Manager'
        args: [pathToOpen]

  showSelectedEntryInFileManager(editor.getPath())


atom.contextMenu.add {
  'atom-text-editor': [{label: 'Show In File Manager', command: 'util:show-file-in-manager'}]
}
```

**Comment by sass94** [comment]
blockquote, div.yahoo_quoted { margin-left: 0 !important; border-left:1px #715FFA solid !important; padding-left:1ex !important; background-color:white !important; } Take me off thread.
Please. Don't know how I got here. 


Sent from Yahoo Mail for iPhone


On Sunday, December 11, 2016, 5:48 PM, Wliu <notifications@github.com> wrote:


Closed #589.
